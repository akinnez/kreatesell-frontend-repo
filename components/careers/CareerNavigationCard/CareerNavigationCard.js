import React from 'react'
import PropTypes from "prop-types";
import Image from "next/image"
import {useRouter} from "next/router";

import {Button} from "../../";
import {LeftArrow} from "../../../utils";
import styles from "./CareerNavigationCard.module.scss"

const CareerNavigationCard = ({department, description, role}) => {
  const router = useRouter();
  return (
    <div className={styles.navigationContainer}>
        <div className={styles.left}>
            <div className={styles.top}>
                <div className={styles.arrow} onClick={()=>router.push("/careers")}>
                  <Image src={LeftArrow} alt="" width="40"/>
                </div>
                <h3 className={styles.heading}>{department}</h3>
            </div>
            <div className={styles.bottom}>
                <h2 className={styles.title}>{role}</h2>
                <p className={styles.description}>{description}</p>
            </div>
        </div>
        <div className={styles.right}>
            <Button text="Apply" bgColor="blue" className={styles.actionBtn}/>
        </div>
    </div>
  )
}

CareerNavigationCard.defaultProps = {

}

CareerNavigationCard.propTypes = {
  department: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired
}

export default CareerNavigationCard;