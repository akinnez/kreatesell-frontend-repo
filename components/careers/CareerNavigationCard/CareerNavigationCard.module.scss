@import '../../../public/css/colors.scss';
@import '../../../public/css/mixins.scss';
@import '../../../public/css/screens.scss';

.navigationContainer {
	background: #e6f7ff;
	height: 230px;
	width: 100%;
	padding: 2rem 10.31rem;
	@include flex(row, space-between, center);
	@include screen('tablet') {
		padding: 0rem 4rem;
	}
	@include screen('medium') {
		padding: 0rem 2rem;
	}
	.left {
		@include screen('medium') {
			height: 100%;
			padding-block-start: 2rem;
		}
		.top {
			@include flex(row, null, center);
			gap: 10px;
			margin-bottom: 15px;
			.arrow {
				@include flex(row, null, center);
				cursor: pointer;
				@include screen('medium') {
					display: none;
				}
			}
			.heading {
				font-weight: 600;
				font-size: 24px;
				line-height: 32px;
				color: #0072ef;
				margin-bottom: 0px;
			}
		}
		.bottom {
			.title {
				font-weight: 700;
				font-size: 28px;
				line-height: 40px;
				color: #262626;
			}
			.description {
				font-weight: 400;
				font-size: 16px;
				line-height: 26px;
				color: #8c8c8c;
			}
			@include screen('tablet') {
				.title {
					font-size: 1.4rem;
				}
				.description {
					font-size: 1rem;
				}
			}
		}
	}

	.right {
		@include screen('medium') {
			height: 100%;
			padding-block-start: 2rem;
		}
		.actionBtn {
			@include flex(row, center, flex-start);
			min-width: 10rem;
			height: 3.125rem;
			align-items: center;
			font-size: 1rem;
			padding-inline: 1.4rem;
		}
	}
}
