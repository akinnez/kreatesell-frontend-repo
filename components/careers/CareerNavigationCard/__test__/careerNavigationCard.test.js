import CareerNavigationCard from '../CareerNavigationCard';
import renderer from 'react-test-renderer';

const mockProps = {
	department: 'IT',
	description: 'mock description',
	role: 'Back-end developer',
};

it('renders a snapshot of <CareerNavigationCard />', () => {
	const card = renderer
		.create(<CareerNavigationCard {...mockProps} />)
		.toJSON();
	expect(card).toMatchSnapshot();
});
