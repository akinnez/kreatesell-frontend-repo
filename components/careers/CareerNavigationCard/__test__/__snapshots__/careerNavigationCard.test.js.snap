// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <CareerNavigationCard /> 1`] = `
<div
  className="navigationContainer"
>
  <div
    className="left"
  >
    <div
      className="top"
    >
      <div
        className="arrow"
        onClick={[Function]}
      >
        <img
          src="[jest.find('../utils')]"
        />
      </div>
      <h3
        className="heading"
      >
        IT
      </h3>
    </div>
    <div
      className="bottom"
    >
      <h2
        className="title"
      >
        Back-end developer
      </h2>
      <p
        className="description"
      >
        mock description
      </p>
    </div>
  </div>
  <div
    className="right"
  >
    <button
      className="
			actionBtn
			button
			false 
			btnBlue
			false
			"
      type="submit"
    >
      <div
        className="buttonValue"
      >
        Apply
      </div>
    </button>
  </div>
</div>
`;
